static/app/utils/replays/hooks/useExtractDiffMutations.tsx

```diff
@@ -0,0 +1,222 @@
+        lastFrame = frame;
+
+        const removes = {};
+        for (const removal of frame.data.removes) {
+          const node = mirror.getNode(removal.id) as HTMLElement | null;
+          if (!node || !node.outerHTML) {
+            continue;
+          }
+          const selector = getSelectorForElem(node);
+          const rootIsRemoved = Object.keys(removes).some(key =>
+            selector.startsWith(key)
+          );
+          if (rootIsRemoved) {
+            continue;
+          }
+
+          removes[selector] = {
+            html: node.outerHTML,
+          };
+        }
+
+        collection.set(frame, {
+          [frame.timestamp]: {
+            adds: {},
+            attributes: {},
+            removes,
+            offset: frame.timestamp - startTimestampMs,
+          },
+        });
+      }
+    },
+  });
+  return results;
+}
+
+function getNameForElem(element: HTMLElement) {
+  if (element.id) {
+    return `#${element.id}`;
+  }
+  const parts = [
+    element.tagName.toLowerCase(),
+    element.className && typeof element.className === 'string'
+      ? `.${element.className.split(' ').filter(Boolean).join('.')}`
+      : '',
+  ];
+  const attrs = [
+    'data-sentry-element',
+    'data-sentry-source-file',
+    'data-test-id',
+    'data-testid',
+  ];
+  for (const attr of attrs) {
+    const value = element.getAttribute(attr);
+    if (value) {
+      parts.push(`[${attr}=${value}]`);
+    }
+  }
+  return parts.join('');
+}
+
+// Copy Selector => `#blk_router > div.tsqd-parent-container > div > div`
+// Copy JS Path => `document.querySelector("#blk_router > div.tsqd-parent-container > div > div")`
+// Copy XPath => `//*[@id="blk_router"]/div[2]/div/div`
+// Copy Full XPath => `/html/body/div[1]/div[2]/div/div`
+function getSelectorForElem(element: HTMLElement): string {
+  const parts: string[] = [];
+  let elem: HTMLElement | null =
+    element.nodeType !== Node.ELEMENT_NODE ? element.parentElement : element;
+
+  while (elem) {
+    parts.unshift(getNameForElem(elem));
+    if (elem.id !== '' || (elem.getAttribute('data-sentry-element') ?? '') !== '') {
+      break;
+    }
+    elem = elem.parentElement;
+  }
+  return parts.join(' > ');
+}
+
+interface Props {
+  leftOffsetMs: number;
+  replay: ReplayReader;
+  rightOffsetMs: number;
+}
+
+export default function useExtractDiffMutations({
+  leftOffsetMs,
+  replay,
+  rightOffsetMs,
+}: Props): UseQueryResult<Map<ReplayFrame, DiffMutation>> {
+  const startTimestampMs = replay.getReplay().started_at.getTime();
+  const rangeStartTimestampMs = startTimestampMs + leftOffsetMs;
+  const rangeEndTimestampMs = startTimestampMs + rightOffsetMs;
+
+  return useQuery({
+    queryKey: [
+      'extractDiffMutations',
+      replay,
+      rangeStartTimestampMs,
+      rangeEndTimestampMs,
+    ],
+    queryFn: () =>
+      extractDiffMutations({replay, rangeStartTimestampMs, rangeEndTimestampMs}),
+    enabled: true,
+    gcTime: 0, // Infinity,
```

Comment:
oops, had this set to `0` for testing because i didn't want the cache to kick in so often.